pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+titled "Using GitLab Issues for CPython".
+
+Why not other issue tracker
+---------------------------
+
+Using another issue tracker will require yet another learning curve, of having
+to learn and get used to a different interface. We'll also need to learn and
+figure out how to build the integrations with GitHub.
+
+By using GitHub issues, where the source code is currently hosted, where pull requests
+are taking place, we'll be providing consistent experience to contributors and
+maintainers, and not having to jump from one interface to another.
+
+If someone else wants to do the research of what it takes for Python to start using
+other issue tracker, and what the migration plan is, please go ahead and come
+back with a PEP.
+
+Why not focus on improving Roundup / bpo
+----------------------------------------
+
+GitHub has many features we like that are already available. We still need to
+build out additional integrations and update our bots, but this is something
+we already know how to do.
+
+In order to really improve Roundup / bpo, it needs to first migrate to GitHub,
+add CI and bots. As I understand it, there is hesitation because upstream Roundup
+is still im mercurial. Someone more familiar with Roundup / bpo needs
+to champion this effort. (I'm not volunteering, I'm sorry).
+
+I believe the effort of creating and maintaining GitHub integrations and bots
+is much less than the effort needed to get Roundup up to speed and then to continue
+maintaining it.
+
+Migration Plan
+==============
+
+Backup of GitHub data
+---------------------
+
+This effort has been started and being tracked as an issue in core-workflow [#backup-github]_.
+We're using GitHub Migrations API [#gh-migrations-api]_ to download GitHub data
+for CPython on daily basis. The archives will be dropped in an S3 bucket.
+
+Thanks to Ernest W. Durbin III for working on this.
+
+Update the CLA host
+-------------------
+
+At the moment the CLA is hosted within bpo. It needs to be updated such that
+signing the CLA does not require a bpo account, and it should be hosted outside
+of the bpo.
+
+The current CLA process itself is not ideal. Currently contributors to
+devguide, peps, and core-workflow need to sign a CLA,  and it requires a bpo
+account. A bpo account should not be required for those projects.
+
+Currently the CLA process involves real human to manually check the records.
+Question: Will it be possible to completely automate the CLA process, so
+it does not require human intervention?
+
+Create labels for issue triage
+------------------------------
+
+In bpo, we currently have the following fields for each issues:
+
+Type: behavior, crash, compile error, resource usage, security, performance, enhancement.
+Components: 2to3, Argument Clinic, asyncio, Build, Cross-build, ctypes, ...
+Priority: release blocker, deferred blocker, critical, high, normal, low
+
+We will create the corresponding labels::
```

Comment:
Hmm, this is a lot of labels. Have you given thought or experimented to see how well the Github interface handles this many labels?

Reply:
I haven't seen any mention of limit of labels on GitHub anywhere. These are labels created but in really not all of them will be applied to the issue/PR. 
In coala we've been working with [65 labels](https://github.com/coala/coala/labels?page=1&sort=name-asc), and we also use bots to automatically categorize [PRs](https://github.com/coala/coala/pulls).

Reply:
Aah perfect, you already have a project that has a lot of labels 👍